### ISSUE TITLE
Label DD Lines in Pro

### ISSUE THREAD

**Issue by dfoll** [Esri/distance-direction-addin-dotnet#542]
### Background
Users need to have labels on features created by Distance and Direction. 

### Current State
We are unable to create labels for any features being created for Distance and Direction in ArcGIS Pro.

### Expected Result

Functionality is implemented
Tests written
Tested on Pro 2.1, 2.2
Tested in Distance and Direction 2.1 and 2.2
Tested in Military Tools 2.1 and 2.2
Documented in Distance and Direction and Military Tools

### Tasks to be done - Last update 25 May 2018
- [x] Fix that Lines are not drawing for Bearing and Distance
https://github.com/Esri/distance-direction-addin-dotnet/issues/576
- [x] Design how the user would expect labels to happen - BB working with DFoll and CMoore on this.
- [x] Verify that layer package matches other tabs (circles is the first tab worked on) 
- [x] Review the manual checklist test - BB (PR for Save As feature/KMZ output https://github.com/ArcGIS/solutions-defense-test-catalog/pull/146)
- [x] Review the workflow
- [x] Update / review the CMS resources notes
- [x] Review the homepage for Military Tools and Distance and Direction 
- [x] Hook up delete all
- [x] Share a list of questions with the team - Chris

- [x] Switch from drawing graphics to drawing features
- [x] Add all feature class types for DD to ToC - Lyle (verify)
- [x] Featureclasses should be inside a featuredataset
- [x] enable labels
- [x] Writing the arcade scripts for label experssions - Bob
- [x] Create 2 layer packages with sample features (schema only and one with data) - Bob - Done in this PR: https://github.com/Esri/distance-direction-addin-dotnet/pull/565

### Verify
- [x] Features are created in default database
- [x] Feature are deleted from default database 
- [x] Labels are using the correct style
- [x] Verify Update documentation

### Background information
location of the layer package
https://github.com/Esri/distance-direction-addin-dotnet/compare/csm/%23541-pro-features

**Comment by BobBooth** [comment]
I have a draft set of layer packages (using Arccade, with schema to support labeling) that I started before the planning meeting, and just wrapped up. Will submit a PR to the repo. @dfoll - would you mind having a look?  I looked at the Circle LPK arcade expression and adapted it to match the ArcMap line style.

**Comment by BobBooth** [comment]
I am taking off the Design tag, on the assumption that matching the labels for DD lines in ArcMap and the arcade expression from the Circle LPKX is correct.

**Comment by csmoore** [comment]
Dev work started in https://github.com/Esri/distance-direction-addin-dotnet/tree/csm/%23542-pro-line-features-labels 

Another issue I noticed with the "Lines of Bearing" layer package (now at: https://github.com/Esri/distance-direction-addin-dotnet/tree/csm/%23541-pro-features/source/addins/ProAppDistanceAndDirectionModule/ProAppDistanceAndDirectionModule/Data ) is that the Geo Lines apparently have Z values so can't currently be stored in this feature class. I can jump through some hoops to removes these Z's but you may want this support.

To summarize the current issues with the ["Lines of Bearing" layer](https://github.com/Esri/distance-direction-addin-dotnet/tree/csm/%23541-pro-features/source/addins/ProAppDistanceAndDirectionModule/ProAppDistanceAndDirectionModule/Data):

1. Differences between lines export datamodel and currently implemented
2. Should support Z's

Moving issue to impeded for team to decide on those 2 issues.

**Comment by BobBooth** [comment]
I reviewed the manual checklist test, added a PR for Save As feature/KMZ output in Cucumber style: https://github.com/ArcGIS/solutions-defense-test-catalog/pull/146

**Comment by csmoore** [comment]
Using the latest layer package - got a little further but hit this familiar problem when using the schema-only lpkx:

![image](https://user-images.githubusercontent.com/3090809/40549528-b49a38ba-6005-11e8-9f04-b62bb47a86c8.png)

The workaround to the problem - the same problem we had here: https://github.com/Esri/military-features-data/issues/287 is to run the Add Spatial Index GP tool prior to packaging the schema-only lpkx. Just not sure how/why this problem still exists.

Adding the workaround - LOBs should look like this now - I'll add a PR now with just the current/working "Distance_And_Direction.lpkx" we should all be working from.

![image](https://user-images.githubusercontent.com/3090809/40550166-59bea6b8-6007-11e8-9bdf-808acab672b4.png)

**Comment by csmoore** [comment]
The dev work for this is currently in PR #571 - removing dev label

**Comment by csmoore** [comment]
Resuming dev to do PR rework mentioned here: https://github.com/Esri/distance-direction-addin-dotnet/pull/571#issuecomment-392126556

**Comment by csmoore** [comment]
PR #571 rework should now be complete - removing dev label

**Comment by topowright** [comment]
Reviewed the documentation and there does not need to be any changes to any workflows related to the lines tab.

**Comment by lfunkhouser** [comment]
The functionality to label lines by distance and bearing will be handled in issue #576  . Removing the dependency and the task to fix this issue from this issue, Label DD Lines in Pro.
